backend/image_analyzer.py

```
@@ -1,109 +1,35 @@
         Returns a Pydantic model (RegularReceipt, TransportationTicket, or NotAReceipt)
         """
         if not os.path.exists(image_path):
             raise FileNotFoundError("Image not found")
         
         try:
-            if self.provider == AIProvider.AZURE.value:
-                return self._analyze_with_azure(image_path)
-            else:
-                return self._analyze_with_gemini(image_path)
+            return self._analyze_image_with_gemini(image_path)
         except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)